c++/include/orc/Reader.hh

```diff
@@ -32,13 +32,40 @@
 #include <set>
 #include <string>
 #include <vector>
+#include <atomic>
 
 namespace orc {
 
   // classes that hold data members so we can maintain binary compatibility
   struct ReaderOptionsPrivate;
   struct RowReaderOptionsPrivate;
 
+  /**
+   * Expose the reader metrics including the latency and
+   * number of calls of the decompression/decoding/IO modules.
+   */
+  struct ReaderMetrics {
+    std::atomic<uint64_t> ReaderCount{0};
+    // ReaderInclusiveLatencyUs contains the latency of
+    // the decompression/decoding/IO modules.
+    std::atomic<uint64_t> ReaderInclusiveLatencyUs{0};
+    std::atomic<uint64_t> DecompCount{0};
+    // DecompLatencyUs contains the IO blocking latency.
+    std::atomic<uint64_t> DecompLatencyUs{0};
+    std::atomic<uint64_t> DecodingCount{0};
+    // DecodingLatencyUs contains the latency of the decompression/IO modules.
```

Comment:
I think we can do better here to get the pure decoding time. Please see the inline comments below.